core/src/main/java/org/aerogear/mobile/core/reactive/CallableRequest.java

```diff
@@ -0,0 +1,98 @@
+package org.aerogear.mobile.core.reactive;
+
+import java.util.concurrent.Callable;
+import java.util.concurrent.atomic.AtomicReference;
+
+import android.util.Log;
+
+import org.aerogear.mobile.core.Request;
+import org.aerogear.mobile.core.Responder;
+import org.aerogear.mobile.core.utils.SanityCheck;
+
+/**
+ * This class will synchronously invoke a callable parameter when
+ * {@link CallableRequest#respondWith(Responder)} is invoked
+ *
+ * @param <T> a type that this request generates
+ */
+public final class CallableRequest<T> extends AbstractRequest<T> {
+
+    private static final String ERROR_TAG = "UNCAUGHT_EXCEPTION";
+    private final Callable<T> callable;
+    private Thread callableThread = null;
+
+    public CallableRequest(Callable<T> callable) {
+        SanityCheck.nonNull(callable, "callable");
+        this.callable = callable;
```

Comment:
this can be `this.callable = nonNull(callable, "callable");`

Reply:
I smell a new lint rule we can implement.

Reply:
You have there that sanity check but we should also have `@NonNull Callable<T> callable` there for it to work with IDE's checks and Kotlin's null-safety.